### PR TITLE
Fix find one cant have multiple parameters when there is a primary key

### LINKED ISSUE #4901: FindOne can't have multiple parameters if it contains primary key with mongoose
**Describe the bug**
When using the strapi-hook-mongoose lib and trying to ```findOne``` on a model, if you're giving a primary key as one of the params, the rest is ignored. 

**Steps to reproduce the behavior**
1. Create a model with a boolean "private" for example
2. Create an entry with private at true
3. From its controller, create a findOne method that will findOne of your model with the ctx.params + ```private: false```
4. Allow this api route as public
5. You can retrieve your entry with the api despite it being a "private" entry

**Expected behavior**
You should be able to have multiple params to a findOne (it's not findById)

**Code snippets**
Source for the findOne in strapi-hook-mongoose:
```
  async function findOne(params, populate) {
    const primaryKey = getPK(params);

    if (primaryKey) {
      // Here params is redefined completely
      params = {
        [model.primaryKey]: primaryKey,
      };
    }

    const entry = await model
      .findOne(params)
      .populate(populate || defaultPopulate);

    return entry ? entry.toObject() : null;
  }
````

**System**

- Node.js version: v11.12.0

- NPM version: 6.7.0

- Strapi version: beta.17.8

- Database: Mongodb

### PR DESCRIPTION
**Pull request by petersg83**
Fix #4901 

#### Description of what you did:
- move primary key logic from `strapi-connector-*` to `strapi-database`
- queries done with all params now, not only the primaryKey when there's one
- destructuring `ctx.params` where it was needed (to avoid unexpected params from koa-router)

### CODE REVIEW

**Comment by derrickmehaffy** [comment]
@petersg83 can you check on the failing DCO check (need to sign commits)

File: packages/strapi-database/lib/utils/primary-key.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+const _ = require('lodash');
+
+module.exports = {
+  replaceIdByPrimaryKey: (params, model) => {
```
Comment (alexandrebodin) [inline packages/strapi-database/lib/utils/primary-key.js]: You will need to create a test file for this new function :)

Just a styling comment: 
```js
module.exports = {
  replaceIdByPrimaryKey(params, model) {
    // code
  }
}

```

**Comment by marxvn** [comment]
Cannot edit content in admin-ui when changing primaryKey

**Comment by petersg83** [comment]
Hi,
Can you please open a new issue for that ?

**Comment by marxvn** [comment]
> Hi,
> Can you please open a new issue for that ?

yep #5741

**Pull request merged**